File: Programmers/네트워크.py
from collections import deque

def dfs(num, visited, graph):
    q = deque()
    q.append(num)
    visited[num] = True
    
    while q :
        node = q.popleft()
        
        for i in graph[node]:
            if not visited[i]:
                visited[i] = True
                q.append(i)
    

def solution(n, computers):
    visited = [False] * (n)
    graph = [[] for _ in range(n)]
    answer = 0
    
    for i in range(len(computers)):
        for j in range(n) :
            if computers[i][j] == 1:
                graph[i].append(j)
    
    for i in range(n):
        if not visited[i]:
            dfs(i, visited, graph)
            answer += 1
    
    return answer

File: Swift/네트워크.swift
import Foundation

func solution(_ n:Int, _ computers:[[Int]]) -> Int {
    
    var visited = Array(repeating: false, count: n)
    var answer = 0
    
    func bfs(_ node: Int) {
        var queue = [node]
        while !queue.isEmpty {
            let newN = queue.popLast()!
            visited[newN] = true
            for i in 0..<computers[newN].count {
                if computers[newN][i] == 1 && !visited[i] {
                    queue.append(i)
                }
            }   
        }
    }
    
    for i in 0..<n {
        if !visited[i] {
            bfs(i)
            answer += 1
        }
    }
    return answer
}